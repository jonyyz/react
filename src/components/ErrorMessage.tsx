import * as React from "react";

import {
  makeStyles,
  createStyles,
} from "@material-ui/styles";
import Typography from "@material-ui/core/Typography";
import ErrorIcon from "@material-ui/icons/Error";

const useStyles = makeStyles(createStyles({
  container: {
    margin: 20,
    display: "flex",
    flexDirection: "row"
  },
  errorMessage: {
    marginTop: 6,
    marginLeft: 6,
  },
}));

interface IProps {
  text: string;
}

const ErrorMessage: React.FunctionComponent<IProps> = React.memo(({ text }) => {
  const classes = useStyles({});

  return (
    <div className={classes.container}>
      <ErrorIcon color="error" fontSize="large" />
      <Typography className={classes.errorMessage} variant="body1">{text}</Typography>
    </div>
  );
});

export default ErrorMessage;
