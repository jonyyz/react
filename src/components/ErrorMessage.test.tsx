import * as React from "react";
import { shallow } from "enzyme";

import ErrorMessage from "./ErrorMessage";

describe("ErrorMessage", () => {
  test("should render", () => {
    const wrapper = shallow(<ErrorMessage text="error text" />);
    expect(wrapper).toMatchSnapshot();
  });
});
