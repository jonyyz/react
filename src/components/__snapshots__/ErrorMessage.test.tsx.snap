// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should render 1`] = `
<div
  className="makeStyles-container-1"
>
  <ErrorIcon
    color="error"
    fontSize="large"
  />
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-errorMessage-2"
    variant="body1"
  >
    error text
  </WithStyles(ForwardRef(Typography))>
</div>
`;
